src/cdk/text-field/autosize.ts

```diff
@@ -222,21 +224,27 @@ export class CdkTextareaAutosize implements AfterViewInit, DoCheck, OnDestroy {
 
     const textarea = this._elementRef.nativeElement as HTMLTextAreaElement;
     const value = textarea.value;
+    const placeholderText = textarea.placeholder;
 
     // Only resize if the value or minRows have changed since these calculations can be expensive.
-    if (!force && this._minRows === this._previousMinRows && value === this._previousValue) {
+    if (!force && this._minRows === this._previousMinRows && value === this._previousValue &&
+        placeholderText === this._previousPlaceholderText) {
       return;
     }
 
-    const placeholderText = textarea.placeholder;
+    const placeholderLongerThanValue = placeholderText.length > value.length;
```

Comment:
Comparing the length will probably work most of the time, but could cause a problem in cases when the characters don't end up being close enough to equal in width (e.g. "I" could be 30% the width of a "W"). 

Since placeholders rarely change, we could alternatively cache the height of the textarea with the placeholder text (invalidating if it changes), and use that as an effective lower-bound for this function.

Reply:
That seems reasonable to me. Just to make sure, you are requesting that I make this change right?

Reply:
If you think it's reasonable then yeah

Reply:
sadly had to go back to the original approach because I couldn't get the caching approach to work with view-engine

Reply:
went back to caching as per offline comments